ParabolInc/parabol
# Add JiraServer integration provider fields

Add required fields to the database and `addIntegrationProvider`
mutation so the provider can be added.

## Testing
* generate public/private key pair and a consumer key
  ```sh
  openssl genrsa -out jira_privatekey.pem 1024
  openssl req -newkey rsa:1024 -x509 -key jira_privatekey.pem -out jira_publickey.cer -days 365
  openssl pkcs8 -topk8 -nocrypt -in jira_privatekey.pem -out jira_privatekey.pcks8
  openssl x509 -pubkey -noout -in jira_publickey.cer >jira_publickey.pem
  openssl rand -base64 32 >jira_consumerkey
  ```
* run the mutation
  ```graphql
  mutation {
    addIntegrationProvider(input: {
      teamId: " paste valid team id "
      service: jiraServer
      authStrategy: oauth1
      scope:org
      oAuth1ProviderMetadataInput: {
        serverBaseUrl: "https://jira.parabol.co/"
        consumerKey: " paste content of jira_consumerkey "
        privateKey: """ paste content of jira_privatekey.pem """
      }
    }) {
      __typename
      ... on AddIntegrationProviderSuccess {
        provider {
          id
          isActive
          teamId
          ... on IntegrationProviderOAuth1 {
            serverBaseUrl
          }
        }
      }
    }
  }
  ```
* verify values in DB

Resolves: #5809
